clang/lib/Sema/SemaModule.cpp

```diff
@@ -650,6 +650,14 @@ DeclResult Sema::ActOnModuleImport(SourceLocation StartLoc,
   else
     VisibleModules.setVisible(Mod, ImportLoc);
 
+  assert((!Mod->isModulePartitionImplementation() || getCurrentModule()) &&
+         "We can only import a partition unit in a named module.");
+  if (Mod->isModulePartitionImplementation() &&
+      getCurrentModule()->isModuleInterfaceUnit())
+    Diag(ImportLoc,
+         diag::warn_import_implementation_partition_unit_in_interface_unit)
+        << Mod->Name;
```

Comment:
Not for this PR, but at some point it might make sense to have an overload for `operator<<` so we can pass a `const Module *` directly and it automatically quotes the name, similar to how `NamedDecl` works.

Reply:
Got it. I'll try to make it next time